package.json

```
@@ -0,0 +1,47 @@
+{
+  "name": "camomile",
+  "version": "0.0.0",
+  "description": "Node.js HTTP image proxy to route images through SSL",
+  "keywords": [
+    "ssl",
+    "node",
+    "proxy",
+    "camo",
+    "camomile",
+    "security",
+    "unified"
+  ],
+  "homepage": "https://unifiedjs.com",
+  "bugs": "https://github.com/rehypejs/camomile/issues",
+  "repository": "rehypejs/camomile",
+  "funding": {
+    "type": "opencollective",
+    "url": "https://opencollective.com/unified"
+  },
+  "license": "MIT",
+  "author": "Merlijn Vos <merlijn@soverin.net>",
+  "type": "module",
+  "exports": "./index.js",
+  "files": [
+    "lib/",
+    "index.d.ts",
+    "index.js"
+  ],
+  "scripts": {
+    "build": "tsc --build --clean && tsc --build",
+    "format": "remark . -qfo && prettier . -w --log-level warn",
+    "test": "npm run build && npm run format && npm run test-api",
+    "test-api": "node --test test.js"
+  },
+  "dependencies": {
+    "ipaddr.js": "^2.1.0",
+    "undici": "^5.24.0"
+  },
+  "devDependencies": {
+    "@types/node": "^18.17.15",
+    "prettier": "^3.0.3",
+    "remark-cli": "^11.0.0",
+    "remark-preset-wooorm": "^9.1.0",
+    "typescript": "^5.2.2"
```

```suggestion
  "dependencies": {
    "ipaddr.js": "^2.0.0",
    "undici": "^5.0.0"
  },
  "devDependencies": {
    "@types/node": "^18.0.0",
    "prettier": "^3.0.0",
    "remark-cli": "^11.0.0",
    "remark-preset-wooorm": "^9.0.0",
    "typescript": "^5.0.0"
```

Use loose version ranges if unknown. It creates cleaner package locks, and makes it easier for package managers to resolve conflicts

---

I don't understand why you would do this. GPT can't come up with the benefits either:

https://chat.openai.com/share/5e4aa769-4e0b-4214-a966-fec85c82cc2f

---

It allows package managers to deduplicate versions.
If a dependency is already being used without issue, there is no reason to force npm/yarn to install two copies of it.

---

That is a trade off not worth making in my opinion. Rounding down may introduce bugs or missing features if a package was at ^2.1.0 instead of ^2.0.0, there's usually a reason for that. Besides we can't guarantee it will dedupe versions in all cases. Sounds like taking risks for the small gain of using less disk space. Which is practically a XY problem, if disk space on your personal computer is a worry there are many ways to deal with that other than making our dependencies less predictable and safe.

---

On its own I don’t really think it matters that much which semver range you choose when starting a project. It does become a problem when updating.

I’ve seen packages update their semver ranges, and make a release for it. Then their dependants do the same thing. This can send a wave of pointless automated pull requests by dependency management bots throughout the ecosystem.

I like to keep things simple and consistent and use `^x.0.0` everywhere.